src/etos_test_runner/lib/testrunner.py

```diff
@@ -106,13 +164,18 @@ def run_tests(self, workspace):
             with Executor(test, self.iut, self.etos) as executor:
                 self.logger.info("Starting test '%s'", executor.test_name)
                 executor.execute(workspace)
-
                 if not executor.result:
                     result = executor.result
-                self.logger.info("Test finished. Result: %s.", executor.result)
-        return result
+                self.logger.info(
+                    "Test finished. Result: %s. Test framework exit code: %d",
+                    executor.result,
+                    executor.returncode,
+                )
+        return result, executor.returncode
 
-    def outcome(self, result, executed, description):
+    def outcome(
```

Comment:
I'm thinking that maybe we should have a `DefaultVerdictMatcher` which deals with the other cases

Pseudo:
```
def __init__(self, iut, etos):
    self.verdict_matcher = self._load_verdict_matcher()
    ...

def _load_verdict_matcher(self):
    custom_rules = os.getenv("VERDICT_RULE_FILE")
    if custom_rules:
        return CustomVerdictMatcher(rules)
    else:
        # We could also just have a VerdictMatcher and instead pass a DEFAULT_RULES dict to it.
        return DefaultVerdictMatcher(None)  # passing None if we want base class, otherwise not passing anything is fine.

def outcome(result, executed, description, exit_code):
    return self.verdict_matcher.evaluate(executed, description, exit_code)    
```

Reply:
Created a draft of DefaultVerdictMatcher in my latest update.

Reply:
I did a draft of what it would look like. We agreed it looked too complicated. The current proposal is the minimal change to solve the current issue.